src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -21,6 +24,9 @@ public static final String reverseWithStringConcat(String string) {
     }
 
     public static final String reverseWithStringBuilder(String string) {
+        if (string == null || string.isEmpty()) {
```

Comment:
doing the same thing I just added a check to see if the string was null or empty